### PR TITLE
[cpp-qt5-client ] Improve CMake

### PR DESCRIPTION
**Pull request by MartinDelille**
Here are some little improvement of the CMake files:
- Don't use globs to make sure any modification in a file will rebuild it
- Link ssl and crypto only for Linux

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@ravinikam @stkrwork @etherealjoy @muttleyxd

### CODE REVIEW

File: samples/client/petstore/cpp-qt5/client/CMakeLists.txt [hunk 1]
```diff
@@ -14,12 +14,25 @@ endif ()
...
+    PFXUserApi.cpp
+    PFXHelpers.cpp
+    PFXHttpRequest.cpp
+    PFXHttpFileElement.cpp
 )
+target_link_libraries(${PROJECT_NAME} PRIVATE Qt5::Core Qt5::Network )
 
-add_library(${PROJECT_NAME} ${SRCS})
-target_link_libraries(${PROJECT_NAME} PRIVATE Qt5::Core Qt5::Network ssl crypto)
+if(UNIX AND NOT APPLE)
```
Comment (etherealjoy) [inline samples/client/petstore/cpp-qt5/client/CMakeLists.txt:33]: what about windows?

Comment (MartinDelille) [reply samples/client/petstore/cpp-qt5/client/CMakeLists.txt:33]: I fixed that here: 5367ed8104b2925ec66fa45e5ed7ea850f76a567

Unfortunately I wasn't able to test on Windows but it shouldn't change the behavior now!

**Comment by wing328** [comment]
Pushed a commit to update the samples. Let's see how that goes.

File: samples/client/petstore/cpp-qt5/client/CMakeLists.txt (hunk 1 above)
Comment (wing328) [reply samples/client/petstore/cpp-qt5/client/CMakeLists.txt]: Let's see if our community (window users) can help us confirm that after merging this PR into master.

**Pull request merged**